### ISSUE TITLE
[HC] woocommerce.com requested on every admin page

### ISSUE THREAD

**Issue by anastas10s-afk** [Automattic/woocommerce-services#2738]
The https://api.woocommerce.com/payment/sift URL is being requested on every page in the admin, adding .67s (depending on the server response time) to the page generation time of admin pages.

The external request to the sift URL is found within the admin_print_footer_scripts action [in the woocommerce-services\woocommerce-services.php file, on line #897](https://github.com/Automattic/woocommerce-services/blob/aefaea37ec441dd710c77b09e37e7383bf8d98e7/woocommerce-services.php#L897):
`add_action( ‘admin_print_footer_scripts’, array( $this, ‘add_sift_js_tracker’ ) );`

This will load the “add_sift_js_tracker” function on every admin page, which then requests https://api.woocommerce.com/payment/sift

This can be seen below in the add_sift_js_tracker function which calls “get_sift_configuration” that makes the HTTP request to the woocomerce.com/payment/sift URL

```
    public function add_sift_js_tracker() {
        $sift_configurations = $this->api_client->get_sift_configuration();
    }
```

```
    public function get_sift_configuration() {
        return $this->request( 'GET', '/payment/sift' );
    }

```

Here is a tool to see the response time of that request from around the world:
https://tools.keycdn.com/performance?url=https://api.woocommerce.com/payment/sift

From Sydney it’s taking .74s to respond . This adds .74s to the backend page generation whilst WordPress waits for the request to finish.

From what I gather, related to #2689 .
Reported at:
- https://wordpress.org/support/topic/woocommerce-com-requested-on-every-admin-page/
- 8217803-zen

**Comment by der-martin85** [comment]
I added these lines to the beginning of the add_sift_js_tracker method in "woocommerce-services.php":

```
                public function add_sift_js_tracker() {
                        global $pagenow;
                        if ($pagenow !== 'admin.php' || !isset($_GET['page']) || !in_array($_GET['page'], ['wc-orders'])) return;
```

So now the tracker is only added to pages where **I think** it is needed.

Maybe this can be solved by the page that requires the tracker to set some global variable that is being checked in the method instead of checking a list of pages in the method.

**Comment by AntonyXSI** [comment]
Is there a reason it needs to load on the orders page? Ideally there wouldn't be any external requests impacting the load time on the order listings page as it will be requested often. If it is required is there any way to cache the api.woocommerce.com/payment/sift get request for a short period or does the nonce, signature etc returned from that request need to be dynamic.

Bear in mind the wc-orders page is only for when HPOS is enabled otherwise the standard post type edit screen is requested (edit.php?post_type=shop_order)

**Comment by der-martin85** [comment]
I'm not 100% sure what this thing is used for, but I guess it's for creating Shipping Labels and on the wc-orders page (or the shop_order page) you can create Shipping Labels.

But, I will leave this conversation as I uninstalled the plugin, because in my search through why we have it in the first place I found out, that we don't need it. We had another Plugin for the Labels.

**Comment by openmindculture** [comment]
If WooCommerce needs to load external scripts, it should use proper WordPress enqueue hooks, so that we can dequeue them at least, or add an option in the WooCommerce settings. 
- What is Sift good for?
- Is it safe to turn it off?
- How to do so without patching plugin code?

**Comment by rferguson10** [comment]
agreed with all of the above. I tried using remove_action to get rid of this to no avail

**Comment by WildWolf-Ryzen** [comment]
Any update to fix this in the near future or maybe provide a code that will only call the API on necessary pages.

**Comment by AntonyXSI** [comment]
Any update or progress on this? I submitted the original request to .org and still see an HTTP request to https://api.woocommerce.com/payment/sift on every admin page when this plugin is installed. 

Surely this doesn't need to be requested on every admin page? Is there a limited scope this can run or be cached?

**Comment by AntonyXSI** [comment]
It looks like the shipping functionality of this plugin has been moved to a separate stand alone plugin "WooCommerce Shipping". Although the shipping part of this plugin still works for users who haven't made the switch yet. 

The new shipping plugin has actually changed the add_sift_js_tracker and get_sift_configuration functions with the Sift lookup request cached in a transient for one month. This stops the requests being made on every admin page and confirms that it can be cached. It would be nice if this was integrated into the WC Tax plugin to help speed up this plugin for users still using this plugin for shipping.

**Comment by momedul** [comment]
It appears you're encountering a performance issue where an unnecessary external API request to `api.woocommerce.com/payment/sift` is being made on every admin page, slowing down page load times. This is likely related to the Sift integration for shipping label generation.

To address this, consider implementing a caching mechanism for the `get_sift_configuration` request. A good approach would be to cache the result of this API call using WordPress transients, similar to how it's handled in the newer "WooCommerce Shipping" plugin. This would prevent repeated requests on every admin page load.

For a deeper understanding of how WooCommerce handles extensibility and its various hooks and filters, which could be relevant for managing such integrations, please refer to this article:

https://needlecode.com/blog/woocommerce/deep-dive-woocommerce-extensibility-hooks-filters.html

**Comment by Abdalsalaam** [comment]
Fixed in #2913.
